.gitignore

```diff
@@ -3,12 +3,13 @@
 debug/
 target/
 
-# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
-# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
-Cargo.lock
 
 # These are backup files generated by rustfmt
 **/*.rs.bk
 
 # MSVC Windows builds of rustc generate these, which store debugging information
 *.pdb
+
+# VS Code and RustRover
+.vscode/
+.idea/
```

Comment:
normaly this shouldn't be here

that being said I'm not against having it in the gitignore but if it is here I'd like it to also include vim's swap files
`*.sw[pom]`

Reply:
Ok, I'll add it asap